### PR TITLE
I126 user card is missing in contact page

### LINKED ISSUE #126: User card is missing in Contact page
**Is your feature request related to a problem? Please describe.**
The user card is missing in Contact page after integrating with Firebase.

**Describe the solution you'd like**
Display the user card at the very top of the contact list in Contact page.
Check figma to see how the user card should be displayed.

### PR DESCRIPTION
**Pull request by scratchclaggy**
## Change Summary
Addresses #126 

This change introduces a new field to the Firestore `userDoc` called info.

image

Essentially, the user's information is stored in an object called info that shares the same shape as a contact. In the contact list, an index is used to navigate to the detailed view / form for a given contact, so this change has modified the way the index is used in two ways:

- The index can now be `null`, this is true when no contact has been selected (i.e. the list view). Prievously the value `-1` was used for this state
- The value `-1` now refers to the the user's information, rather than one of their contacts

Additionally, a new callback was exposed from the Firestore context to allow updating of the user's info.

### Change Form

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
 
These changes will require a DB wipe / schema change.

# Related Issue

- Resolve #126

### CODE REVIEW

**Review by skyheat — COMMENTED** [review]
Looks good to me but I'm not too familiar with the firebase specifics. I'll let @NicholasChoong take a look at it.

File: src/components/Contact/contactitem.tsx [hunk 1]
```diff
@@ -22,12 +23,11 @@ const ContactItem = ({ contact, image, firestoreIndex }: ContactItemProps) => {
...
           <Avatar image={image} height={40} width={40} iconClass='h-10 w-10' />
-          <p className='font-medium text-gray-700'>
-            {contact.clientName}
-            {/* {contacts[firestoreIndex].clientName} */}
-          </p>
+          <p className='font-medium text-gray-700'>{contact.clientName}</p>
         </span>
-        <p className='text-gray-500'>{truncateText(contact.pets, 16)}</p>
+        {contact.pets && (
+          <p className='text-gray-500'>{truncateText(contact.pets, 16)}</p>
```
Comment (NicholasChoong) [inline src/components/Contact/contactitem.tsx:29]: we can also truncate text with css
https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

File: src/context/ContactContext/index.tsx [hunk 2]
```diff
@@ -45,7 +47,7 @@ const ContactProvider = ({ children }: { children: ReactNode }) => {
           return false
         }
 
-        if (reg.test(contact.clientName) || reg.test(contact.pets)) {
+        if (reg.test(contact.clientName) || reg.test(contact.pets!)) {
```
Comment (NicholasChoong) [inline src/context/ContactContext/index.tsx:50]: should we check if pets is not null first before passing it to another function?

File: src/pages/contact.tsx [hunk 3]
```diff
@@ -41,18 +45,19 @@ const Contact = () => {
...
+                Add
+              </Button>
             </div>
-          )}
-          {getDisplayContactIndex() === -1 && !getCreatingNewContact() && (
+          </div>
+        )}
+        <div className='m-auto max-w-md'>
+          {getDisplayContactIndex() === null && !getCreatingNewContact() && (
             <div className='border-grey m-2 flex flex-row rounded-xl border-2'>
```
Comment (NicholasChoong) [inline src/pages/contact.tsx:61]: maybe call the getters and store the values in some constants before the jsx to reduce the number of function calls

**Review by NicholasChoong — CHANGES_REQUESTED** [review]
Great work

**Review by NicholasChoong — APPROVED** [review]
Good Work

**Pull request merged**